sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -465,13 +465,13 @@ case class AlterTableAddPartitionCommand(
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
-    val table = catalog.getTableMetadata(tableName)
+    val table = catalog.getTableRawMetadata(tableName)
```

Comment:
do we need to make this change? `PartitioningUtils.normalizePartitionSpec` will restore the raw schema of partition columns.